lib/SearchProvider.js

```diff
@@ -134,7 +132,7 @@ NewTabSearchProvider.prototype = {
   /**
    *  Gets the state - a combination of the current engine and all the visible engines.
    */
-  asyncGetCurrentState: Task.async(function*() {
+  get currentState() {
```

Comment:
Or actually, we could just keep the Task.async and have it be `function()` instead of `function*()`